integrationTests/__fixtures__/passing/jest-runner-eslint.config.js

```diff
@@ -1,5 +1,8 @@
+const { ESLint } = require('eslint');
+
 module.exports = {
   cliOptions: {
-    global: ['hello'],
+    // `ESLint` requires this to be an object, not an array
+    global: ESLint ? { hello: true } : ['hello'],
```

Comment:
this is main reason I think this is a breaking change

Reply:
thinking about it, #125 _might_ be considered breaking as well, if people previously ignored an engine warning and used this package on node 6. I don't consider it a breaking change, but bundling these together in a new major removes any ambiguity 🤷

Reply:
It seems pretty valuable to try to support both eslint 8 and < 8 at the same time

Reply:
sure, but I don't think it's our responsibility to coerce user input into the correct type. Consumers should configure the the version of ESLint they choose to use correctly

Reply:
Oh true, fair point - i think it’s 100% fine if a user switches to eslint 8, and this runner breaks until they also fix their configuration (altho hopefully they get a very clear error message)

Reply:
Yeah, the error message from eslint is pretty good 👍  Specifically in this case if I leave it as an array

```
ESLint configuration in CLIOptions is invalid:
      - Property "globals" is the wrong type (expected object but got `["hello"]`).
```